packages/devtools_shared/lib/src/test/integration_test_runner.dart

```diff
@@ -0,0 +1,335 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
+
+import 'chrome_driver.dart';
+import 'io_utils.dart';
+
+const _testSuffix = '_test.dart';
+
+class IntegrationTestRunner with IOMixin {
+  static const _beginExceptionMarker = 'EXCEPTION CAUGHT';
+  static const _endExceptionMarker = '═════════════════════════';
+  static const _errorMarker = ': Error: ';
+  static const _unhandledExceptionMarker = 'Unhandled exception:';
+  static const _allTestsPassed = 'All tests passed!';
+  static const _maxRetriesOnTimeout = 1;
+
+  Future<void> run(
+    String testTarget, {
+    required String testDriver,
+    bool headless = false,
+    List<String> dartDefineArgs = const <String>[],
+    bool debugLogging = false,
+  }) async {
+    void debugLog(String log) {
+      if (debugLogging) print(log);
+    }
+
+    Future<void> runTest({required int attemptNumber}) async {
+      debugLog('starting the flutter drive process');
+      final process = await Process.start(
+        'flutter',
+        [
+          'drive',
+          // Debug outputs from the test will not show up in profile mode. Since
+          // we rely on debug outputs for detecting errors and exceptions from the
+          // test, we cannot run this these tests in profile mode until this issue
+          // is resolved.  See https://github.com/flutter/flutter/issues/69070.
+          // '--profile',
+          '--driver=$testDriver',
+          '--target=$testTarget',
+          '-d',
+          headless ? 'web-server' : 'chrome',
+          for (final arg in dartDefineArgs) '--dart-define=$arg',
```

Comment:
The only change to this class is to how these arguments are added and the parameter list for this method changed slightly. This used to be part of run_test.dart above.